### PR TITLE
feat(search): Add Search Functionality to Resource List in Insights Page

### PR DESCRIPTION
**Pull request by Cookiery**
## What type of PR is this?
/kind feature
/kind chore

## What this PR does / why we need it:

- Add Search Functionality to Resource List in Insights Page
- Update Makefile to use full path for swag and swagger commands

## Which issue(s) this PR fixes:

https://github.com/KusionStack/karpor/issues/679

### CODE REVIEW

**Comment by elliotxx** [comment]
Your movement is fast, let me take a look at these modifications.

**Review by MDA1422 — COMMENTED** [review]
Are there any specific scenarios or edge cases that we should test for the keyword search functionality?

**Review by MDA1422 — COMMENTED** [review]
Is the updated documentation compliant with any particular markdown rendering requirements or guidelines?

**Comment by Cookiery** [comment]
> Is the updated documentation compliant with any particular markdown rendering requirements or guidelines?

Yes, the doc is generated by `make gen-api-doc`, but when i submitted it, I found that there were several places that needed manual line breaks, or needed to add `<br>` in the comments.

**Comment by Cookiery** [comment]
> Are there any specific scenarios or edge cases that we should test for the keyword search functionality?

The only other character that could affect it is spaces, so I can add a line to remove spaces. I don't have any other ideas. Initially, I thought about preventing SQL injection, but the input query is already an SQL statement.

File: pkg/core/handler/search/search.go [hunk 1]
```diff
@@ -71,6 +75,13 @@ func SearchForResource(searchMgr *search.SearchManager, aiMgr *ai.AIManager, sea
 		if searchPage <= 1 {
 			searchPage = 1
 		}
+		if searchKeyword != "" && searchQuery != "" {
```
Comment (elliotxx) [inline pkg/core/handler/search/search.go:78]: @Cookiery  You can add a todo comment here, and consider compatibility with keywords when there are more search patterns in the future

Comment (Cookiery) [reply pkg/core/handler/search/search.go:78]: Isn't this interface only for the search box? If not, the query above needs to be changed as well.

Comment (elliotxx) [reply pkg/core/handler/search/search.go:78]: Yes, this interface is currently used for the resource search box on the homepage and insight page, but it will support multiple search patterns in the future, such as DSL/Cybernetes

**Review by elliotxx — APPROVED** [review]
lgtm
